api/client/credentials.go

```diff
@@ -194,6 +218,14 @@ func (c *identityCredsFile) SSHClientConfig() (*ssh.ClientConfig, error) {
 	return sshConfig, nil
 }
 
+// Expiry returns the credential expiry.
+func (c *identityCredsFile) Expiry() (time.Time, bool) {
+	if err := c.load(); err != nil {
```

Comment:
I'm not super convinced by this. I copied this code from the other functions but it doesn't look thread-safe 🫠 .
We might want to add a lock or something.

Reply:
👍  IIRC there isn't any need for it to be thread-safe currently, as the two methods that call it are called sequentially, but it could definitely use a lock to prevent future issues.

Reply:
done